test/rtl/basic.mlir

```diff
@@ -3,43 +3,43 @@
 // CHECK-LABEL: func @test1(%arg0: i3) -> i50 {
 func @test1(%arg0: i3) -> i50 {
   // CHECK-NEXT:    %c42_i12 = rtl.constant(42 : i12) : i12
-  // CHECK-NEXT:    %[[RES0:.*]] = rtl.add %c42_i12, %c42_i12 : i12
-  // CHECK-NEXT:    %[[RES1:.*]] = rtl.mul %c42_i12, %[[RES0:.*]] : i12
+  // CHECK-NEXT:    [[RES0:%[0-9]+]] = rtl.add %c42_i12, %c42_i12 : i12
+  // CHECK-NEXT:    [[RES1:%[0-9]+]] = rtl.mul %c42_i12, [[RES0]] : i12
   %a = rtl.constant(42 : i12) : i12
   %b = rtl.add %a, %a : i12
   %c = rtl.mul %a, %b : i12
 
-  // CHECK-NEXT:    %[[RES2:.*]] = rtl.sext %arg0 : (i3) -> i7
-  // CHECK-NEXT:    %[[RES3:.*]] = rtl.zext %arg0 : (i3) -> i7
+  // CHECK-NEXT:    [[RES2:%[0-9]+]] = rtl.sext %arg0 : (i3) -> i7
+  // CHECK-NEXT:    [[RES3:%[0-9]+]] = rtl.zext %arg0 : (i3) -> i7
```

Comment:
Why not use patterns for function arguments too?

Reply:
because there is no need to.  The arguments are deterministicly numbered.

Writing tests is an art, and more patterns isn't necessarily better - it is a cost/benefit ratio between maintainability over time and the cost to produce the test.  Using fixed names like %arg0 is "fine" since it isn't going to produce a maintenance problem over time.

Reply:
Ok I see, thanks for explaining!